src/test/java/com/hazelcast/kubernetes/ServiceEndpointResolverTest.java

```diff
@@ -46,127 +43,117 @@
 
-    @Test
-    public void resolveWithNamespaceAndNodeInNamespaceWithNoSubsets() {
-        Endpoints discoveryNode = createEndpoints(1);
-        discoveryNode.setSubsets(null);
-        nodesInNamespace.getItems().add(discoveryNode);
+        ServiceEndpointResolver sut = new ServiceEndpointResolver(LOGGER, null, 0, SERVICE_LABEL, SERVICE_LABEL_VALUE, NAMESPACE,
+                null, KUBERNETES_MASTER_URL, API_TOKEN);
 
-        ServiceEndpointResolver sut = new ServiceEndpointResolver(LOGGER, SERVICE_NAME, 0, null, null, NAMESPACE, null, KUBERNETES_MASTER_URL, API_TOKEN);
+        // when
         List<DiscoveryNode> nodes = sut.resolve();
 
-        assertEquals(0, nodes.size());
+        // then
+        assertEquals(1, nodes.size());
+        assertEquals(2, nodes.get(0).getPrivateAddress().getPort());
     }
 
     @Test
-    public void resolveWithServiceLabelAndNodeInNamespace() {
-        nodesInNamespace.getItems().add(createEndpoints(1));
+    public void resolveWithServiceNameWhenNotReadyAddressesAndNotReadyEnabled() {
+        // given
+        Endpoints endpoints = createNotReadyEndpoints(2);
+        given(client.endpointsByName(NAMESPACE, SERVICE_NAME)).willReturn(endpoints);
 
-        ServiceEndpointResolver sut = new ServiceEndpointResolver(LOGGER, SERVICE_NAME, 0, SERVICE_LABEL, SERVICE_LABEL_VALUE, NAMESPACE, null, KUBERNETES_MASTER_URL, API_TOKEN);
-        List<DiscoveryNode> nodes = sut.resolve();
-
-        assertEquals(0, nodes.size());
-    }
+        ServiceEndpointResolver sut = new ServiceEndpointResolver(LOGGER, SERVICE_NAME, 0, null, null, NAMESPACE,
+                RESOLVE_NOT_READY_ADDRESSES, KUBERNETES_MASTER_URL, API_TOKEN);
 
-    @Test
-    public void resolveWithServiceLabelAndNodeWithServiceLabel() {
-        nodesInNamespace.getItems().add(createEndpoints(1));
-        Endpoints discoveryNode = createEndpoints(2);
-        nodesWithLabel.getItems().add(discoveryNode);
-
-        ServiceEndpointResolver sut = new ServiceEndpointResolver(LOGGER, SERVICE_NAME, 0, SERVICE_LABEL, SERVICE_LABEL_VALUE, NAMESPACE, null, KUBERNETES_MASTER_URL, API_TOKEN);
+        // when
         List<DiscoveryNode> nodes = sut.resolve();
 
+        // then
         assertEquals(1, nodes.size());
-        assertEquals(2, nodes.get(0).getPrivateAddress().getPort());
     }
 
     @Test
-    public void resolveWithNamespaceAndNotReadyAddresses() {
-        nodesInNamespace.getItems().add(createEndpoints(1));
-        nodesInNamespace.getItems().add(createNotReadyEndpoints(2));
+    public void resolveWithServiceNameWhenNotReadyAddressesAndNotReadyDisabled() {
+        // given
+        Endpoints endpoints = createNotReadyEndpoints(2);
+        given(client.endpointsByName(NAMESPACE, SERVICE_NAME)).willReturn(endpoints);
+
+        ServiceEndpointResolver sut = new ServiceEndpointResolver(LOGGER, SERVICE_NAME, 0, null, null, NAMESPACE, null,
```

Comment:
This would seriously benefit from a builder pattern (in general, I don't mean in this PR btw!)

Reply:
I can't agree more. But definitely not part of this PR, it's already big.